taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -468,43 +471,56 @@ func (self *Contract) Run(ctx vm.CallFrame, evm *vm.EVM) ([]byte, error) {
 			fmt.Println("Unable to parse getValidators input args: ", err)
 			return nil, err
 		}
-
-		res := self.getValidators(args)
-		return method.Outputs.Pack(res.Validators, res.End)
+		return method.Outputs.Pack(self.getValidators(args))
 
 	case "getDelegations":
 		var args sol.GetDelegationsArgs
 		if err = method.Inputs.Unpack(&args, input); err != nil {
 			fmt.Println("Unable to parse getDelegations input args: ", err)
 			return nil, err
 		}
-
-		res := self.getDelegations(args)
-		return method.Outputs.Pack(res.Delegations, res.End)
+		return method.Outputs.Pack(self.getDelegations(args))
 
 	case "getUndelegations":
 		var args sol.GetUndelegationsArgs
 		if err = method.Inputs.Unpack(&args, input); err != nil {
 			fmt.Println("Unable to parse getUndelegations input args: ", err)
 			return nil, err
 		}
-
-		res := self.getUndelegations(args)
-		return method.Outputs.Pack(res.Undelegations, res.End)
+		return method.Outputs.Pack(self.getUndelegations(args))
 	}
 
 	return nil, nil
 }
 
-func (self *Contract) DistributeRewards(rewardsStats *rewards_stats.RewardsStats, feesRewards *FeesRewards) {
+func (self *Contract) DistributeRewards(author *common.Address, rewardsStats *rewards_stats.RewardsStats, feesRewards *FeesRewards) {
 	// When calling DistributeRewards, internal structures must be always initialized
 	self.lazy_init()
 
 	// Calculates number of tokens to be generated as block reward
 	blockReward := bigutil.Mul(self.amount_delegated, self.yield_percentage)
 	blockReward = bigutil.Div(blockReward, bigutil.Mul(Big100, self.blocks_per_year))
 
+	voteReward := bigutil.Big0
+	authorReward := bigutil.Big0
+	trxReward := blockReward
+	// We need to handle case for block 0
```

Comment:
how about we simply dont call DistributeRewards for block 0 ?

Reply:
this is wrong comment, even block 1 does not have any votes (reward votes) and you should do the distribution on that block

Reply:
This is a question, how can a question be wrong comment ? :)

Reply:
This is a question, how can a question be wrong comment ? :)

Reply:
and why should you ? It is totally up to us, if we decide not to distribute rewards for first 2 blocks... No need to have rewards for genesis block at all and I am pretty sure we could live also without rewarding block 1 as our algorithm uses votes for rewarding, which are not yet present in block 1, totally valid

Reply:
I meant `// We need to handle case for block 0` is wrong comment

Reply:
yes, we can skip reward distribution for first 2 blocks, but what will you do with trxs fees. This is of course for bigger discussion but I would distribute them

Reply:
I guess it is up to us, we can have this if here or we can simply not distribute rewards for first 2 blocks...

Reply:
true